docs/markdown/Builtin-options.md

```diff
@@ -113,6 +115,21 @@ the two-way mapping:
 
 All other combinations of `debug` and `optimization` set `buildtype` to `'custom'`.
 
+#### Details for `warning_level`
+
+Exact flags per warning level is compiler specific, but there is an approximative
+table for most common compilers.
+
+| Warning level | GCC/Clang                | MSVC  |
```

Comment:
Do you think there is a better name for the second column? I presume there are other compilers that Meson supports that just piggyback on GCC/Clang arguments.

Reply:
And they declare themselves as get_argument_syntax() == 'gcc'. :)

Reply:
I guess I always thought argument syntax was more so the command line argument style, not necessarily an indication of supporting the exact same arguments. Not like it matters anyways

Reply:
We can probably extend that table for other compilers if they don't follow GCC, but for now those are the main ones.

Reply:
IMHO the naming is good like that, if someone has better column name to suggest please do :)

Reply:
> more so the command line argument style, not necessarily an indication of supporting the exact same arguments. 

Correct, but in practice they mean the same thing. :D

For human readable documentation, "like GCC" is more useful than "exactly GCC" since stuff like -Wall -Wextra all tend to be "family arguments", not "compiler specific arguments".